### ISSUE TITLE
Unable to establish a %any IPsec IKEv2 tunnel with multiple subnet

### ISSUE THREAD

**Issue by siuswat** [libreswan/libreswan#275]
After I installed Libreswan 3.28. I get TS_UNACCEPTABLE error in pluto logs and tunnel gets established for only one subnet pair. When I change right=%any to right=<Remote IP> in config, both tunnel gets established. 
Same config works with IKEv1 and Libreswan 3.20 with IKEv2

conn Tunnel-LAN
  left=1.2.3.4
  leftsubnet=192.168.26.0/23
  leftid=192.168.2.1
  right=%any
  rightsubnet=192.168.32.0/22
  rightid=192.168.32.62
  ike=aes128-sha1-modp2048
  ikelifetime=86400s
  phase2alg=aes128-sha1;modp2048
  keylife=14400s
  ikev2=yes
  dpddelay=30
  dpdtimeout=120
  dpdaction=clear
  authby=secret
  auto=add

conn Tunnel-DMZ
  left=1.2.3.4
  leftsubnet=192.168.2.0/23
  leftid=192.168.2.1
  right=%any
  rightsubnet=192.168.32.0/22
  rightid=192.168.32.62
  ike=aes128-sha1-modp2048
  ikelifetime=86400s
  phase2alg=aes128-sha1;modp2048
  keylife=14400s
  ikev2=yes
  dpddelay=30
  dpdtimeout=120
  dpdaction=clear
  authby=secret
  auto=add

Error log from pluto log:
responding to CREATE_CHILD_SA message (ID 2) from <Remote IP>:4500 with encrypted notification TS_UNACCEPTABLE

**Comment by letoams** [comment]
I've replicated this with test case ikev2-06-dual-net

**Comment by logbob0401** [comment]
This issue also happen when use left or rightsubnets to hold multiple subnets

**Comment by letoams** [comment]
note the default changed from ikev1 to ikev2. It might be you just need to add ikev2=no to your config to make it work like before

**Comment by siuswat** [comment]
Both sides are running the same version when upgrading from 3.20 to 3.28. And the same multiple subnet configuration works on Libreswan 3.20 with IKEv2

**Comment by jempatel** [comment]
The same issue has been observed when host to host tunnel is created with nat. It gets the same message of TS_UNACCEPTABLE.

Adding ikev2=no in config solves the issue.

**Comment by cagney** [comment]
looks fixed in mainline; old from 3.30 test results:
```
1v2 "westnet-eastnet-ipv4-psk-ikev2" #1: initiating IKEv2 IKE SA
1v2 "westnet-eastnet-ipv4-psk-ikev2" #1: STATE_PARENT_I1: sent v2I1, expected v2R1
1v2 "westnet-eastnet-ipv4-psk-ikev2" #2: STATE_PARENT_I2: sent v2I2, expected v2R2 {auth=IKEv2 cipher=AES_GCM_16_256 integ=n/a prf=HMAC_SHA2_512 group=MODP2048}
002 "westnet-eastnet-ipv4-psk-ikev2" #2: IKE_AUTH response contained the error notification TS_UNACCEPTABLE
000 "westnet-eastnet-ipv4-psk-ikev2" #2: scheduling retry attempt 1 of an unlimited number, but releasing whack
```
new from mainline:
```
 ipsec auto --up westnet-eastnet-ipv4-psk-ikev2
1v2 "westnet-eastnet-ipv4-psk-ikev2" #1: initiating IKEv2 connection
1v2 "westnet-eastnet-ipv4-psk-ikev2" #1: sent IKE_SA_INIT request to 192.1.2.23:500
1v2 "westnet-eastnet-ipv4-psk-ikev2" #1: sent IKE_AUTH request {cipher=AES_GCM_16_256 integ=n/a prf=HMAC_SHA2_512 group=MODP2048}
003 "westnet-eastnet-ipv4-psk-ikev2" #1: initiator established IKE SA; authenticated peer using authby=secret and ID_FQDN '@east'
004 "westnet-eastnet-ipv4-psk-ikev2" #2: initiator established Child SA using #1; IPsec tunnel [192.0.1.0-192.0.1.255:0-65535 0] -> [192.0.2.0-192.0.2.255:0-65535 0] {ESP/ESN=>0xESPESP <0xESPESP xfrm=AES_GCM_16_256-NONE DPD=passive}
west #
 ipsec auto --up westnet-eastnet-ipv4-psk-ikev2-b
002 "westnet-eastnet-ipv4-psk-ikev2-b" #3: initiating Child SA using IKE SA #1
1v2 "westnet-eastnet-ipv4-psk-ikev2-b" #3: sent CREATE_CHILD_SA request for new IPsec SA
004 "westnet-eastnet-ipv4-psk-ikev2-b" #3: initiator established Child SA using #1; IPsec tunnel [192.0.1.0-192.0.1.255:0-65535 0] -> [192.0.3.0-192.0.3.255:0-65535 0] {ESP/ESN=>0xESPESP <0xESPESP xfrm=AES_GCM_16_256-NONE-MODP2048 DPD=passive}
west #
```
